transaction/std/src/transaction_builder.rs

```diff
@@ -162,6 +174,17 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
             .push(InputMaterials::Signable(input_credentials));
     }
 
+    /// Add a View Only Input to the transaction.
+    ///
+    /// # Arguments
+    /// * `input_view_only_materials` - Materials required to construct the
+    ///   onetime private key for the input when combined with the spend private
+    ///   key for the account.
```

Comment:
I think this is a bit confusing since when using this particular API I am not sure when and how the spend private key will be provided, if at all. Isn't the idea here to use this to generate signing data that can then later be combined with a spend private key to produce a signed Tx?

Reply:
yes, that is the intention. Currently, there is no default way to provide the spend private key outside of the changes that are coming to the transaction signer and UnsignedTx within the FullService repo.